### PR TITLE
[SPARK-4556] [BUILD] binary distribution assembly can't run in local mode

### PR DESCRIPTION
**Pull request by srowen**
Add note on building a runnable distribution with make-distribution.sh

### CODE REVIEW

File: docs/building-spark.md [hunk 1]
```diff
@@ -34,6 +34,14 @@ and in `project/SparkBuild.scala` add:
...
 to the `sharedSettings` val. See also [this PR](https://github.com/apache/spark/pull/2883/files) if you are unsure of where to add these lines.
 
+# Building a Runnable Distribution
+
+To create a Spark distribution like those distributed by the 
+[Spark Downloads](http://spark.apache.org/downloads.html) page, and that is laid out so as 
+to be runnable, use `make-distribution.sh` in the project root directory. It can be configured 
+with Maven profile settings and so on like the direct Maven build. Example:
+
+    ./make-distribution.sh --name custom-spark --tgz -Phadoop-2.4 -Pyarn
```
Comment (busbey) [inline docs/building-spark.md]: add a closing "for more information on usage run `./make-distribution.sh --help`"

**Comment by andrewor14** [comment]
@pwendell do we want to document the `make-distribution` script explicitly? I remember there's some discussion on that.

**Comment by pwendell** [comment]
Yeah I think so - we've had this around for a long time and I use it heavily as part of releases. I've actually made some effort to support backwards compatibility in this script over the years, so IMO it's something I'm prepared to keep supporting.

**Comment by vanzin** [comment]
This is great, but we also use the opportunity to get rid of that "bigtop-dist" profile altogether? It generates broken artifacts, as shown in the bug.

**Comment by busbey** [comment]
Does that profile get used by BigTop for packaging spark downstream?

**Comment by vanzin** [comment]
> Does that profile get used by BigTop for packaging spark downstream?

Good question; but if they use it, they must have a lot of extra code to deal with the broken tarball it generates. But yes, maybe better to leave that for a separate change and give the bigtop guys a heads up.

**Pull request closed without merging**